plugin/keyring_vault/vault_key.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MYSQL_VAULT_KEY_H
+#define MYSQL_VAULT_KEY_H
+
+#include "keyring_key.h"
+#include "i_serialized_object.h"
+
+namespace keyring {
+
+struct Vault_key : public Key, public ISerialized_object
+{
+  Vault_key(const char *a_key_id, const char *a_key_type, const char *a_user_id,
+      const void *a_key, size_t a_key_len)
+    : Key(a_key_id, a_key_type, a_user_id, a_key, a_key_len)
+    , was_key_retrieved(FALSE)	
+  {}
+
+  Vault_key(const Vault_key &vault_key)
```

Comment:
Auto generated copy constructor would only made a shallow copy of a Key and what I need is a deep one. In Key I did not need a copy constructor, but for Vault_key I need one, as Vault_key can be a serialized_object.